node/src/types/block.rs

```diff
@@ -1265,51 +1265,25 @@ impl Item for Block {
 
 /// A wrapper around `Block` for the purposes of fetching blocks by height in linear chain.
 #[derive(Clone, Debug, PartialOrd, Ord, PartialEq, Eq, Hash, Serialize, Deserialize)]
-pub enum BlockByHeight {
-    Absent(u64),
-    Block(Box<Block>),
+pub struct BlockWithMetadata {
+    pub block: Block,
+    pub finality_signatures: BlockSignatures,
 }
 
-impl From<Block> for BlockByHeight {
-    fn from(block: Block) -> Self {
-        BlockByHeight::new(block)
-    }
-}
-
-impl BlockByHeight {
-    /// Creates a new `BlockByHeight`
-    pub fn new(block: Block) -> Self {
-        BlockByHeight::Block(Box::new(block))
-    }
-
-    pub fn height(&self) -> u64 {
-        match self {
-            BlockByHeight::Absent(height) => *height,
-            BlockByHeight::Block(block) => block.height(),
-        }
-    }
-}
-
-impl Display for BlockByHeight {
+impl Display for BlockWithMetadata {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        match self {
-            BlockByHeight::Absent(height) => write!(f, "Block at height {} was absent.", height),
-            BlockByHeight::Block(block) => {
-                let hash: BlockHash = block.header().hash();
-                write!(f, "Block at {} with hash {} found.", block.height(), hash)
-            }
-        }
+        write!(f, "{:?}", self)
```

Comment:
Our `Display` impls should be more concise than the `Debug` ones, so this should be using the `Display` impl for `Block` and `BlockSignatures`.